### PR TITLE
Documentation: Managed resources section

### PR DESCRIPTION
**Pull request by muvaf**
### Description of your changes

Managed resource section for new documentation.

### Checklist

I have:
- [ ] Run `make reviewable` to ensure this PR is ready for review.
- [ ] Ensured this PR contains a neat, self documenting set of commits.
- [ ] Updated any relevant [documentation] and [examples].
- [ ] Reported all new error conditions into the log or as an event, as
  appropriate.

For more about what we believe makes a pull request complete, see our
[definition of done].

[documentation]: https://github.com/crossplane/crossplane/tree/master/docs
[examples]: https://github.com/crossplane/crossplane/tree/master/cluster/examples
[definition of done]: https://github.com/crossplane/crossplane/tree/master/design/one-pager-definition-of-done.md

### CODE REVIEW

**Comment by kasey** [comment]
this is the only single markdown in the `docs/` directory -- should it go into one of the nested directories?

**Comment by hasheddan** [comment]
@kasey I think it might be okay to get stuff merged and then restructure :+1:

File: docs/managed-resources.md [hunk 1]
```diff
@@ -0,0 +1,256 @@
...
+```
+
+In this example, the user provided only a set of labels to select a `ResourceGroup`
+managed resource that already exists in the cluster via `resourceGroupNameSelector`.
+Then after a specific `ResourceGroup` is selected, `resourceGroupNameRef` is filled
+with the name of that `ResourceGroup` managed resource. Then in the last step,
+Crossplane fills the actual `resourceGroupName` field with whatever format Azure
+accepts it.
+
+Users are able to give the input in all three levels: selector to select via labels,
```
Comment (prasek) [inline docs/managed-resources.md:204]:
```suggestion
Users are able to specify any of these three fields: 
 - selector to select via labels
 - reference to point to a determined managed resource 
 - actual value that will be submitted to the provider
```

File: docs/managed-resources.md [hunk 2]
```diff
@@ -0,0 +1,256 @@
...
+reference to point to a determined managed resource and actual value that will
```
Comment (prasek) [inline docs/managed-resources.md:205]:
```suggestion
```

File: docs/managed-resources.md [hunk 3]
```diff
@@ -0,0 +1,256 @@
...
+be submitted to the provider.
```
Comment (prasek) [inline docs/managed-resources.md:206]:
```suggestion
```

File: docs/managed-resources.md [hunk 4]
```diff
@@ -0,0 +1,256 @@
...
+
+It's important to note that in case a reference exists, the managed resource does
+not create the external resource until the referenced object is ready. In this
+example, creation call of Azure MySQL Server will not be made until referenced
+`ResourceGroup` has its condition named `Ready` to be true.
```
Comment (prasek) [inline docs/managed-resources.md:211]:
```suggestion
`ResourceGroup` has its `status.condition` named `Ready` to be true.
```

File: docs/managed-resources.md [hunk 5]
```diff
@@ -0,0 +1,256 @@
...
+
+## Importing Existing Resources
+
+If you have some resources that are already provisioned in the provider, you can
```
Comment (prasek) [inline docs/managed-resources.md:215]:
```suggestion
If you have some resources that are already provisioned in your cloud provider, you can
```

File: docs/managed-resources.md [hunk 6]
```diff
@@ -0,0 +1,256 @@
...
+spec:
+  providerRef:
+    name: gcp-creds
+```
+
+Crossplane will check whether a GCP Network called `existing-network` exists, and
+if it does, then the optional fields under `forProvider` will be filled with the
+values that are fetched from the provider.
+
+Note that if a resource has required fields, you will need to fill those fields, too,
```
Comment (prasek) [inline docs/managed-resources.md:238]:
```suggestion
Note that if a resource has required fields, you must fill those fields or
```

File: docs/managed-resources.md [hunk 7]
```diff
@@ -0,0 +1,256 @@
...
+because otherwise creation of the managed resource will be rejected. So, in those cases,
```
Comment (prasek) [inline docs/managed-resources.md:239]:
```suggestion
the creation of the managed resource will be rejected. So, in those cases,
```

File: docs/managed-resources.md [hunk 8]
```diff
@@ -0,0 +1,256 @@
...
+you will need to enter the name of the resource as well as values of those required
```
Comment (prasek) [inline docs/managed-resources.md:240]:
```suggestion
you will need to enter the name of the resource as well as all required fields as indicated in the [API Reference][api-reference] documentation.
```

File: docs/managed-resources.md [hunk 9]
```diff
@@ -0,0 +1,256 @@
...
+fields.
+
+## Backup and Restore
+
+Crossplane adheres to Kubernetes conventions as much as possible and one of the
+advantages we gain is backup & restore ability with tools that work with native
+Kubernetes types.
```
Comment (prasek) [inline docs/managed-resources.md:247]:
```suggestion
Kubernetes types, like [Verero](https://velero.io/).
```

File: docs/managed-resources.md [hunk 10]
```diff
@@ -0,0 +1,256 @@
...
+
+If you'd like to backup and restore manually, you can simply export them and save YAMLs
+in your file system. When you reload them, as we've discovered in import section,
+their external nem annotation and requried fields are there and those are enough to
```
Comment (prasek) [inline docs/managed-resources.md:251]:
```suggestion
their `crosssplane.io/external-name` annotation and required fields are there and those are enough to
```

**Comment by muvaf** [comment]
I've addressed @prasek 's comments. @negz should be ready to go.

**Pull request merged**